crates/sui-graphql-rpc/src/types/transaction_block/tx_lookups.rs

```diff
@@ -0,0 +1,468 @@
+            || matches!((self.at_checkpoint, self.before_checkpoint), (Some(at), Some(before)) if at >= before)
+            // If SystemTx, sender if specified must be 0x0. Conversely, if sender is 0x0, kind must be SystemTx.
+            || matches!((self.kind, self.sign_address), (Some(kind), Some(signer)) if (kind == TransactionBlockKindInput::SystemTx) != (signer == SuiAddress::from(NativeSuiAddress::ZERO)))
+    }
+}
+
+/// Determines the maximum value in an arbitrary number of Option<u64>.
+fn max_option<T: Ord>(xs: impl IntoIterator<Item = Option<T>>) -> Option<T> {
+    xs.into_iter().flatten().max()
+}
+
+/// Determines the minimum value in an arbitrary number of Option<u64>.
+fn min_option<T: Ord>(xs: impl IntoIterator<Item = Option<T>>) -> Option<T> {
+    xs.into_iter().flatten().min()
+}
+
+/// Constructs a `RawQuery` as a join over all relevant side tables, filtered on their own filter
+/// condition, plus optionally a sender, plus optionally tx/cp bounds.
+pub(crate) fn subqueries(filter: &TransactionBlockFilter, tx_bounds: TxBounds) -> Option<RawQuery> {
+    let sender = filter.sign_address;
+
+    let mut subqueries = vec![];
+
+    if let Some(f) = &filter.function {
+        subqueries.push(match f {
+            FqNameFilter::ByModule(filter) => match filter {
+                ModuleFilter::ByPackage(p) => (select_pkg(p, sender, tx_bounds), "tx_calls_pkg"),
+                ModuleFilter::ByModule(p, m) => {
+                    (select_mod(p, m.clone(), sender, tx_bounds), "tx_calls_mod")
+                }
+            },
+            FqNameFilter::ByFqName(p, m, n) => (
+                select_fun(p, m.clone(), n.clone(), sender, tx_bounds),
+                "tx_calls_fun",
+            ),
+        });
+    }
+    if let Some(kind) = &filter.kind {
+        subqueries.push((select_kind(*kind, sender, tx_bounds), "tx_kinds"));
+    }
+    if let Some(recv) = &filter.recv_address {
+        subqueries.push((select_recipient(recv, sender, tx_bounds), "tx_recipients"));
+    }
+    if let Some(input) = &filter.input_object {
+        subqueries.push((select_input(input, sender, tx_bounds), "tx_input_objects"));
+    }
+    if let Some(changed) = &filter.changed_object {
+        subqueries.push((
+            select_changed(changed, sender, tx_bounds),
+            "tx_changed_objects",
+        ));
+    }
+    if let Some(sender) = &sender {
+        if filter.requires_explicit_sender() {
+            subqueries.push((select_sender(sender, tx_bounds), "tx_senders"));
+        }
+    }
+    if let Some(txs) = &filter.transaction_ids {
+        subqueries.push((select_ids(txs, tx_bounds), "tx_digests"));
+    }
+
+    let Some((mut subquery, _)) = subqueries.pop() else {
+        return None;
+    };
+
+    if !subqueries.is_empty() {
+        subquery = query!("SELECT tx_sequence_number FROM ({}) AS initial", subquery);
+        while let Some((subselect, alias)) = subqueries.pop() {
```

Comment:
oo i set it up this way because if the name of the tables change it's just ever so slightly easier to adjust them all at the end instead of in the middle of the sandwich...

Reply:
ok you're right, swapping